### PR TITLE
Feat: Implement Inverse PCA function

### PR DESCRIPTION
**Pull request by tc-mateus**
### Description

Implemented a new method for inverse PCA, similar to[ sklearn.decomposition.PCA.inverse_transform()](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html#sklearn.decomposition.PCA.inverse_transform)

### CODE REVIEW

**Review by YuhanLiin — APPROVED** [review]

**Comment by YuhanLiin** [comment]
Can you update to latest master?

File: algorithms/linfa-reduction/src/pca.rs [hunk 1]
```diff
@@ -180,6 +180,14 @@ impl Pca<f64> {
     pub fn singular_values(&self) -> &Array1<f64> {
         &self.sigma
     }
+
+        /// Transform data back to its original space
+    pub fn inverse_transform(
+            &self,
+            prediction: ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>,
```
Comment (YuhanLiin) [inline algorithms/linfa-reduction/src/pca.rs:187]: You can use `Array2<f64>`

File: algorithms/linfa-reduction/src/pca.rs [hunk 2]
```diff
@@ -180,6 +180,14 @@ impl Pca<f64> {
...
+        ) -> ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>> {
```
Comment (YuhanLiin) [inline algorithms/linfa-reduction/src/pca.rs:188]: Also `Array2<f64>`

**Comment by bytesnake** [comment]
Thank you!

**Pull request merged**